base/linalg/eigen.jl

```diff
@@ -155,7 +155,9 @@ function eig(A::AbstractMatrix, B::AbstractMatrix)
     F.values, F.vectors
 end
 function eig(A::Number, B::Number)
-    F = eigfact(A,B)
+    a = fill(A,1,1)
+    b = fill(B,1,1)
+    F = eigfact(a,b)
```

Comment:
I don't think Tony's suggestion is opposition to that. Instead of wrapping the `Number`s in arrays in the `eig` method you could just define
```julia
eigfact(a::Number,b::Number) = eigfact(fill(a,1,1),fill(b,1,1))
```
and the `eig` method would work in it's present version.